tests/lib/rules/no-wait-for-side-effects.test.ts

```diff
@@ -163,7 +163,14 @@ ruleTester.run(RULE_NAME, rule, {
     // fireEvent
     {
```

Comment:
Can you add some valid test cases too? So we check `waitFor` with an arrow function returning a call expression that shouldn't be reported. A couple of cases like these should be fine:

```javascript
`await waitFor(() => somethingElse());`
```

```javascript
{
  settings: { 'testing-library/utils-module': 'test-utils' },
  code: `
    import { waitFor } from 'somewhere-else';
    await waitFor(() => fireEvent.keyDown(input, {key: 'ArrowDown'}))
  `,
},
```

Reply:
From my understanding, this case should pass as invalid. It does not, because `isUserEventUtil()` returns false. Am I missing something? 🤔 

```
    {
      settings: { 'testing-library/utils-module': '~/test-utils' },
      code: `
        import { waitFor, userEvent } from '~/test-utils';
        await waitFor(() => userEvent.click(button))
      `,
      errors: [{ line: 4, column: 29, messageId: 'noSideEffectsWaitFor' }],
    },
```

Reply:
`userEvent` comes from a different package so it's not affected by the Aggressive Reporting mechanism, so it would be only reported if imported from `@testing-library/user-event`

Reply:
Added more test cases in e991eed